app/models/apple/publisher.rb

```diff
@@ -35,27 +35,50 @@ def podcast
       public_feed.podcast
     end
 
-    def episodes_to_sync
-      filter_episodes(show.episodes)
-    end
-
-    def filter_episodes(eps)
+    def filter_episodes_to_sync(eps)
       # Reject episodes if the audio is marked as uploaded/complete
       # or if the episode is a video
       eps
         .reject(&:synced_with_apple?)
         .reject(&:video_content_type?)
+    end
+
+    def episodes_to_sync
+      # only look at the private delegated delivery feed
+      filter_episodes_to_sync(show.apple_private_feed_episodes)
+    end
+
+    def filter_episodes_to_archive(eps)
+      eps_to_sync = Set.new(episodes_to_sync)
+
+      # Episodes to archive can include:
+      # - episodes that are now excluded from the feed
+      # - episodes that are deleted or unpublished
+      # - episodes that have fallen off the end of the feed (Feed#display_episodes_count)
+      eps
+        .reject { |ep| eps_to_sync.include?(ep) }
+        .reject(&:apple_new?)
         .reject(&:archived?)
     end
 
+    def episodes_to_archive
+      # look at the global list of episodes, not just the private feed
+      filter_episodes_to_archive(show.podcast_episodes)
```

Comment:
Okay, and here we're looking at any episode in the podcast that we've already synced to Apple, and archiving them if they fell out of the private feed.